libs/pq_proto/src/framed.rs

```diff
@@ -63,9 +63,9 @@ impl<S> Framed<S> {
         &self.stream
     }
 
-    /// Extract the underlying stream.
-    pub fn into_inner(self) -> S {
-        self.stream
```

Comment:
Instead of returning read buffer, maybe we can try to use `chain` (https://docs.rs/tokio/latest/i686-unknown-linux-gnu/tokio/io/trait.AsyncReadExt.html#method.chain)? To reconstruct the read stream as if the bytes have not been yet read.

Reply:
First swoop failed because Chain (apart from being not public) doesn't implement `AsyncWrite`, truncating away another half.